File: net/watermark/Watermark_add.cs
namespace aspose_snippets.net
{
    public static partial class Watermark
    {
        public static void add()
        {
            const string pathSource = "../../TestData/test.pdf";
            const string watermarkSource = "../../TestData/test.jpg";
            var doc = new Aspose.Pdf.Document(pathSource);

            var artifact = new Aspose.Pdf.WatermarkArtifact();
            artifact.SetImage(new FileStream(watermarkSource, FileMode.Open));

            artifact.ArtifactHorizontalAlignment = Aspose.Pdf.HorizontalAlignment.Center;
            artifact.ArtifactVerticalAlignment = Aspose.Pdf.VerticalAlignment.Center;
            artifact.Rotation = 15;
            artifact.Opacity = 1;
            artifact.IsBackground = true;
            doc.Pages[1].Artifacts.Add(artifact);

            //save result pdf to file
            doc.Save("add_watermark.pdf", Aspose.Pdf.SaveFormat.Pdf);
        }
    }
}

File: java/watermark/src/aspose_snippets/java/Watermark_add.java
package aspose_snippets.java;

public class Watermark_add {
    public static void Execute() {
		        String pathSource = "../../TestData/test.pdf";
            String watermarkSource = "../../TestData/test.jpg";
            var doc = new com.aspose.pdf.Document(pathSource);

            var artifact = new com.aspose.pdf.WatermarkArtifact();
            artifact.setImage(watermarkSource);

            artifact.setArtifactHorizontalAlignment(com.aspose.pdf.HorizontalAlignment.Center);
            artifact.setArtifactVerticalAlignment(com.aspose.pdf.VerticalAlignment.Center);
            artifact.setRotation(15);
            artifact.setOpacity(1);
            artifact.setBackground(true);
            doc.getPages().get_Item(1).getArtifacts().add(artifact);

            //save result pdf to file
            doc.save("add_watermark.pdf", com.aspose.pdf.SaveFormat.Pdf);
    }
}
